plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1397,6 +1404,47 @@ public void setColumnComment(ConnectorSession session, ConnectorTableHandle tabl
         return viewToken;
     }
 
+    @Override
+    public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName tableName)
+    {
+        return tableRedirectionProvider.redirectTable(session, tableName);
+    }
+
+    @Override
+    public Stream<TableColumnsMetadata> streamTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        requireNonNull(prefix, "prefix is null");
+        List<SchemaTableName> schemaTableNames;
+        if (prefix.getTable().isEmpty()) {
+            schemaTableNames = catalog.listTables(session, prefix.getSchema());
+        }
+        else {
+            SchemaTableName schemaTableName = prefix.toSchemaTableName();
+            try {
+                catalog.loadTable(session, schemaTableName);
+                schemaTableNames = ImmutableList.of(schemaTableName);
+            }
+            catch (TableNotFoundException | UnknownTableTypeException e) {
```

Comment:
Why `UnknownTableTypeException`? document

Reply:
I missed this while adapting to support Glue. Hive lists only Iceberg tables.
Yes, we need here `UnknownTableTypeException` handling as well.